doc/tutorial/creating-applications.md

```diff
@@ -0,0 +1,183 @@
+# Creating applications
+
+First of all, let's create an application, which will be the basis of our future
+server API. The [`Application`][application] is the core high-level abstraction
+of the framework. Every application can consist of a number of interfaces,
+each of which has its methods.
+
+For this tutorial, we will create a pretty straightforward chat application
+with two interfaces: `auth` and `chat`. The API will be quite plain with an
+expected scenario like this:
+
+1. User registers a username and password if there is no "account" yet
+2. User logins with valid credentials
+3. Then the user can send messages to all connected users and receive messages
+   from others
+
+For the sake of simplicity we won't use a real database here for storing
+users' credentials or any cryptographic techniques for security, which you
+probably would implement in real projects, because our primary goal is to show
+the basic principles of using JSTP and don't get distracted by implementing
+a lot of other unrelated functionality.
+
+Let's create a `chatApp.js` file in our project directory:
+
+```javascript
+'use strict';
+
+const jstp = require('@metarhia/jstp');
+
+const errors = {
+  ERR_NOT_AUTHORIZED: 1000,
+  ERR_ALREADY_REGISTERED: 1001,
+  ERR_INVALID_CREDENTIALS: 1002,
+  ERR_ALREADY_AUTHENTICATED: 1003,
+};
+
+const users = new Map();
+
+const auth = {
+  register: (connection, username, password, callback) => {
+    if (users.has(username)) {
+      callback(errors.ERR_ALREADY_REGISTERED);
+      return;
+    }
+
+    users.set(username, password);
+    callback(null);
+  },
+  login: (connection, username, password, callback) => {
+    if (connection.session.get('user')) {
+      callback(errors.ERR_ALREADY_AUTHENTICATED);
+      return;
+    }
+
+    const pass = users.get(username);
+
+    if (!pass || pass !== password) {
+      callback(errors.ERR_INVALID_CREDENTIALS);
+      return;
+    }
+
+    connection.session.set('user', username);
+    callback(null);
+  },
+};
+
+const chat = {
+  sendMessage: (connection, message, callback) => {
+    const username = connection.session.get('user');
+
+    if (!username) {
+      callback(errors.ERR_NOT_AUTHORIZED);
+      return;
+    }
+
+    const { server } = connection;
+
+    for (const conn of server.getClients()) {
+      if (conn !== connection) {
+        conn.emitRemoteEvent('chat', 'message', [username, message]);
+      }
+    }
+
+    callback(null);
```

Comment:
IIUC, I should remove `null` in other similar cases too? I just want to emphasize that that method ends without error, with `null` this has seemed clearer

Reply:
Yeah, I talked about 'null', sorry for such unclear message.

I think it's okay to have null there but usually people (at least myself) don't explicitly pass null in case of missing error. 
So it's up to you whether to change it or not. I'll be okay with both.